src/goto-symex/goto_trace.cpp

```
@@ -360,92 +374,141 @@ void correctness_graphml_goto_trace(
   graph.generate_graphml(options);
 }
 
+void goto_tracet::update_coverage(const goto_trace_stept& step) const 
+{
+  if (!step.pc->location.is_nil()) {
+    std::string file = step.pc->location.get_file().as_string();
+    std::string function = step.pc->location.get_function().as_string();
+    std::string line = step.pc->location.get_line().as_string();
+    
+    if (!file.empty() && !line.empty()) {
+      int line_num = std::stoi(line);
+      
+      // Track file coverage
+      auto& file_coverage = coverage_data[file];
+      file_coverage.name = file;
+      file_coverage.covered_lines.insert(line_num);
+      
+      // Track function coverage
+      if (!function.empty()) {
+        auto& func_coverage = file_coverage.functions[function];
+        func_coverage.name = function;
+        func_coverage.file = file;
+        func_coverage.covered_lines.insert(line_num);
+        func_coverage.line_hits[line_num]++;
+        
+        // Update function bounds
+        if (func_coverage.start_line == 0 || line_num < func_coverage.start_line) {
+          func_coverage.start_line = line_num;
+        }
+        if (line_num > func_coverage.end_line) {
+          func_coverage.end_line = line_num;
+        }
+      }
+    }
+  }
+}
+
 void show_goto_trace(
   std::ostream &out,
   const namespacet &ns,
   const goto_tracet &goto_trace)
 {
-  unsigned prev_step_nr = 0;
-  bool first_step = true;
+  try {
```

Shouldn't we just propagate the error instead?